cartoframes/analysis/grid.py

```diff
@@ -1,27 +1,33 @@
 import mercantile
-from shapely.geometry import box
 import pandas as pd
-from cartoframes import CartoDataFrame
+
+from shapely.geometry import box
+from ..core.cartodataframe import CartoDataFrame
 
 
 class QuadGrid():
 
     def polyfill(self, input_gdf, zoom_level):
+        if not hasattr(input_gdf, 'geometry'):
+            raise ValueError('This dataframe has no valid geometry.')
 
-        dfs = []
+        geometry_name = input_gdf.geometry.name
 
-        for _, row in input_gdf.iterrows():
-            bounds = row['geometry'].bounds
+        dfs = []
+        for index, row in input_gdf.iterrows():
+            input_geometry = input_gdf.geometry[index]
+            bounds = input_geometry.bounds
```

Comment:
Instead of doing this how about: 

```
bounds = row[geometry_name].bounds
```
```

Reply:
I have just implemented this change :)

We must keep `input_geometry` because it is used later